lib/charms/prometheus_pushgateway_k8s/v0/pushgateway.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details
+
+r"""Interface library for the Prometheus Pushgateway.
+
+This library wraps the relation endpoint using the `pushgwateway` interface
+and provides a Python API for sending metrics.
+
+
+## Getting Started
+
+To get started using the library, you just need to fetch the library using `charmcraft`.
+
+```shell
+cd some-charm
+charmcraft fetch-lib charms.prometheus_pushgateway_k8s/v0/pushgateway
+```
+
+In the `metadata.yaml` of the charm, add the following:
+
+```yaml
+requires:
+  pushgateway:
+    interface: pushgateway
+```
+
+In the source of your charm, first import the interface:
+
+```
+from charms.prometheus_pushgateway_k8s.v0.pushgateway import PrometheusPushgatewayInterface
+````
+
+Just instantiate the object in `__init__` and then use it when desired to send a metric, passing
+its name and value:
+
+```
+self.ppi = PrometheusPushgatewayInterface(self)
```

Comment:
let's not encourage people to use acronyms when naming variables :) 
`self.prometheus_pushgateway` or `self.pushgateway` are better recommendations to have in the docs.

Reply:
Used `pushgateway_requirer`, thanks!